frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx

```diff
@@ -1,208 +1,340 @@
+  const [selectedTeamId, setSelectedTeamId] = useState<number | null>(
+    parseInt(location?.query?.team_id, 10) || null
+  );
+  const [selectedPolicyIds, setSelectedPolicyIds] = useState<
+    number[] | never[]
+  >([]);
   const [showAddPolicyModal, setShowAddPolicyModal] = useState(false);
   const [showRemovePoliciesModal, setShowRemovePoliciesModal] = useState(false);
-  const [selectedIds, setSelectedIds] = useState<number[] | never[]>([]);
-
+  const [showInheritedPolicies, setShowInheritedPolicies] = useState(false);
   const [updateInterval, setUpdateInterval] = useState<string>(
-    "update interval"
+    "osquery detail update interval"
   );
-  const [policies, setPolicies] = useState<IPolicy[] | never[]>([]);
-  const [isLoading, setIsLoading] = useState(true);
-  const [isLoadingError, setIsLoadingError] = useState(false);
+  // ===== local state
 
-  const getPolicies = useCallback(async () => {
-    setIsLoading(true);
+  const getGlobalPolicies = useCallback(async () => {
+    setIsLoadingGlobalPolicies(true);
+    setIsGlobalPoliciesError(false);
+    let result;
     try {
-      const response = await policiesAPI.loadAll();
-      setPolicies(response.policies);
+      result = await globalPoliciesAPI
+        .loadAll()
+        .then((response) => response.policies);
+      setGlobalPolicies(result);
     } catch (error) {
       console.log(error);
-      setIsLoadingError(true);
+      setIsGlobalPoliciesError(true);
     } finally {
-      setIsLoading(false);
+      setIsLoadingGlobalPolicies(false);
     }
-  }, [dispatch]);
+    return result;
+  }, []);
 
-  const getInterval = useCallback(async () => {
+  const getTeamPolicies = useCallback(async (teamId) => {
+    setIsLoadingTeamPolicies(true);
+    setIsTeamPoliciesError(false);
+    let result;
     try {
-      const response = await configAPI.loadAll();
-      const interval = secondsToHms(
-        inMilliseconds(response.update_interval.osquery_policy) / 1000
-      );
-      setUpdateInterval(interval);
+      result = await teamPoliciesAPI
+        .loadAll(teamId)
+        .then((response) => response.policies);
+      setTeamPolicies(result);
     } catch (error) {
       console.log(error);
-      dispatch(
-        renderFlash(
-          "error",
-          "Sorry, we could not retrieve your update interval."
-        )
-      );
+      setIsTeamPoliciesError(true);
+    } finally {
+      setIsLoadingTeamPolicies(false);
     }
-  }, [dispatch]);
+    return result;
+  }, []);
 
-  useEffect(() => {
-    getPolicies();
-    getInterval();
-  }, [getInterval, getPolicies]);
+  const getPolicies = useCallback(
+    (teamId) => {
+      return teamId ? getTeamPolicies(teamId) : getGlobalPolicies();
+    },
+    [getGlobalPolicies, getTeamPolicies]
+  );
 
-  useEffect(() => {
-    dispatch(queryActions.loadAll());
-  }, [dispatch]);
+  const handleChangeSelectedTeam = useCallback(
```

Comment:
This function seems to be controlled well. Is this function causing the page to re-render multiple times? If not, we don't need `useCallback`.

Reply:
Because handleChangeSelectedTeam is being called in a useEffect hook elsewhere, we need to wrap it in useCallback here.

Reply:
Since it's in the `useEffect`, that means it's being called when needed. If there are no unnecessary re-renders due to that function, then we don't need `useCallback`?

Reply:
Here's the linter warning: 
<img width="920" alt="Screen Shot 2021-09-30 at 6 02 45 PM" src="https://user-images.githubusercontent.com/73313222/135541054-56ef8cbd-e78b-43aa-b532-fd3d3d265750.png">

Reply:
I see. I think we discussed this earlier in the week about a function as a dependency of a `useEffect`. I know there's a linter warning for that too but that's one I ignore and don't agree with because, conceptually, it doesn't make sense in some cases. For example, if I have 

```typescript
useEffect(() => {
  add(val1, val2);
}, [val1, val2])
```

I avoid using `add` as a dependency because I only care about `val1` and `val2` changing. The linter would throw a warning, but it doesn't realize I don't care about it. Similarly, it doesn't seem `handleChangeSelectedTeam` needs to be watched either because it's looking at `router` for a change. `router` would probably never change so the function would never change either.

---

Take a look at this [answer on Stackoverflow](https://stackoverflow.com/a/62601621). It demonstrates someone using `useEffect` and `useCallback` when automatically regenerating markup based on a `count`. This one is a better use-case because the function is changing based on `count`, which is automatically detected based on the number of `sections` given.

For your use-case, the only detection that matters is the team id, which only changes when the user makes it change. If a user changes their team once, then your `useEffect` should only run once. Therefore, `handleChangeSelectedTeam` is well-controlled and doesn't need to be watched. I'm not saying this is incorrect, but it makes the code more expensive because it's checking for differences that generally should never exist. 

So my goal here is to us to help each other understand the best use-cases for our code. I know we're pressed for time, so if you can't do any more testing then we can leave it as it is. But in the future, try comparing the results of using a function as a dependency with a callback vs not. In my experience, the result is the same.

Reply:
Thank you for the detailed explanation. I took a look at the code and I see what you mean about it being well-controlled. I've removed `handleChangeSelectedTeam` from the `useCallback` hook and also as a dependency to the `useEffect` hook. And I'll be pulling the `useQuery` changes we discussed into a new issue. If everything else looks good, I'll get this PR merged as soon as you approve. Thanks again!